### PR TITLE
Add files enrichment using VT data

### LINKED ISSUE #47: Files data enrichment

### PR DESCRIPTION
**Pull request by niedziol**
Closes #47

### CODE REVIEW

File: capstone/files/enrichment/virustotal.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    import requests
+    from requests.auth import HTTPBasicAuth
+except Exception as e:
+    print("No module 'requests' found. Install: pip install requests")
+    sys.exit(1)
+
+try:
+    import vt
+except Exception as e:
+    print("No module 'vt' found. Install pip install vt-py")
```
Comment (mgmacias95) [inline capstone/files/enrichment/virustotal.py:27]: You can check the module's name in the exception object:

```python
>>> try:
...     import blablabla
... except ImportError as e:
...     print(e.name)
... 
blablabla
```

So you can refactor all imports in a single try-except block

File: capstone/files/enrichment/virustotal.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+    # Request VirusTotal info
+    msg = request_virustotal_info(json_alert, apikey)
+
+    # If positive match, send event to Wazuh Manager
+    if msg:
+        send_event(msg, json_alert["agent"])
+
+
+def debug(msg):
```
Comment (mgmacias95) [inline capstone/files/enrichment/virustotal.py:77]: This logger is very basic, this is a good chance to learn about real python loggers. You'd need to configure two logger handlers:
* A [`BaseRotating`](https://docs.python.org/3/library/logging.handlers.html#baserotatinghandler) one.
* A [`Stream`](https://docs.python.org/3/library/logging.handlers.html#streamhandler) which will only be enabled if the `DEBUG` flag is set.

File: capstone/files/enrichment/virustotal.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    if DEBUG:
+        msg = "{0}: {1}\n".format(now, msg)
+
+        print(msg)
+
+        with open(LOG_FILE_PATH, "a") as log_file:
+            log_file.write(msg)
+
+
+def in_database(data, hash):
```
Comment (mgmacias95) [inline capstone/files/enrichment/virustotal.py:87]: Is this used?

**Comment by niedziol** [comment]
One TODO is left. We need to decide when the file is marked as `malicious`.

File: capstone/files/enrichment/virustotal.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+    # Read args
+    alert_file_location = args[1]
+    apikey = args[2]
+
+    logging.info("# API Key")
+    logging.info(apikey)
+
+    logging.info("# File location")
+    logging.info(alert_file_location)
```
Comment (karpp) [inline capstone/files/enrichment/virustotal.py:65]: Same, do you need it in logs?

File: capstone/files/enrichment/virustotal.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        json_alert = json.load(alert_file)
+
+    logging.info("# Processing alert")
+    logging.info(json_alert)
+
+    # Request VirusTotal info
+    msg = request_virustotal_info(json_alert, apikey)
+
+    # If positive match, send event to Wazuh Manager
+    if msg:
```
Comment (karpp) [inline capstone/files/enrichment/virustotal.py:80]: Log error on else

File: capstone/files/enrichment/virustotal.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        return file_info
+    else:
+        alert_output = {}
+        alert_output["virustotal"] = {}
+        alert_output["integration"] = "virustotal"
+
+        logging.error("# Error when conecting VirusTotal API")
+        alert_output["virustotal"]["description"] = "Error: API request fail"
+        send_event(alert_output)
+        exit(0)
```
Comment (karpp) [inline capstone/files/enrichment/virustotal.py:105]: It's not obvious that `get_file_info` can exit program on fail. Mark it on docstring or change to raise/return None

**Review by karpp — CHANGES_REQUESTED** [review]

**Review by karpp — DISMISSED** [review]
LGTM

**Review by karpp — APPROVED** [review]
LGTM

**Review by mgmacias95 — COMMENTED** [review]
update wazuh rules

**Review by mgmacias95 — APPROVED** [review]

**Pull request merged**